### PR TITLE
feat(go-sdk): Helper for retrieving SLI resources

### PR DESCRIPTION
**Pull request by warber**
closes: https://github.com/keptn/keptn/issues/8869

This PR adds functionality to the code base of go-sdk to easily retrieve SLI configuration from Keptn.
Usage:

```go
func (g *EchoHandler) Execute(k sdk.IKeptn, event sdk.KeptnEvent) (interface{}, *sdk.Error) {
...
	sliHelper := keptn.NewSLIHelper(keptn.NewResourceHelper(k.APIV2().Resources()))
	slis, err := sliHelper.GetSLIs(context.TODO(), keptn.GetSLIOptions{
		Project:     "my-project",
		Stage:       "my-stage",
		Service:     "my-service",
		SLIFileName: "prometheus/sli.yaml",
	})
...
}
´´´

### CODE REVIEW

File: pkg/sdk/helpers/resource_client.go [hunk 1]
```diff
@@ -0,0 +1,279 @@
...
+			log.WithFields(
+				log.Fields{
+					projectFieldName:     project,
+					stageFieldName:       stage,
+					serviceFieldName:     service,
+					resourceURIFieldName: resourceURI,
+				}).Debug("Resource not available for service")
+		} else if err != nil {
+			return "", err
+		} else {
```
Comment (bacherfl) [inline pkg/sdk/helpers/resource_client.go:131]: that `else` can be removed

File: pkg/sdk/helpers/resource_client.go [hunk 2]
```diff
@@ -0,0 +1,279 @@
...
+		if errors.As(err, &rnfErrorType) {
+			log.WithFields(
+				log.Fields{
+					projectFieldName:     project,
+					stageFieldName:       stage,
+					resourceURIFieldName: resourceURI,
+				}).Debug("Resource not available for stage")
+		} else if err != nil {
+			return "", err
+		} else {
```
Comment (bacherfl) [inline pkg/sdk/helpers/resource_client.go:154]: that `else` can be removed

File: pkg/sdk/helpers/resource_client.go [hunk 3]
```diff
@@ -0,0 +1,279 @@
...
+		keptnResourceContent, err := rc.GetProjectResource(ctx, project, resourceURI)
+		if errors.As(err, &rnfErrorType) {
+			log.WithFields(
+				log.Fields{
+					projectFieldName:     project,
+					resourceURIFieldName: resourceURI,
+				}).Debug("Resource not available for project")
+		} else if err != nil {
+			return "", err
+		} else {
```
Comment (bacherfl) [inline pkg/sdk/helpers/resource_client.go:175]: that `else` can be removed

File: pkg/sdk/helpers/resource_client.go (hunk 1 above)
Comment (warber) [reply pkg/sdk/helpers/resource_client.go:131]: @bacherfl it cannot be removed, as in this case we would return from the method execution and not check for stage or project resources.

File: pkg/sdk/helpers/resource_client.go (hunk 2 above)
Comment (warber) [reply pkg/sdk/helpers/resource_client.go:154]: same argument as for https://github.com/keptn/go-utils/pull/541#discussion_r984254925

File: pkg/sdk/helpers/resource_client.go (hunk 3 above)
Comment (warber) [reply pkg/sdk/helpers/resource_client.go:175]: same argument as for https://github.com/keptn/go-utils/pull/541#discussion_r984254925

**Review by bacherfl — APPROVED** [review]

**Pull request merged**